main/src/main/java/org/mobilitydata/gtfsvalidator/util/GeospatialUtil.java

```diff
@@ -0,0 +1,163 @@
+/*
+ * Copyright 2021 MobilityData IO
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.mobilitydata.gtfsvalidator.util;
+
+import static org.locationtech.spatial4j.context.SpatialContext.GEO;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Set;
+import org.locationtech.spatial4j.distance.DistanceCalculator;
+import org.locationtech.spatial4j.distance.DistanceUtils;
+import org.locationtech.spatial4j.shape.Point;
+import org.locationtech.spatial4j.shape.Shape;
+import org.locationtech.spatial4j.shape.ShapeFactory;
+import org.locationtech.spatial4j.shape.SpatialRelation;
+import org.mobilitydata.gtfsvalidator.notice.StopTooFarFromTripShapeNotice;
+import org.mobilitydata.gtfsvalidator.table.GtfsLocationType;
+import org.mobilitydata.gtfsvalidator.table.GtfsShape;
+import org.mobilitydata.gtfsvalidator.table.GtfsStop;
+import org.mobilitydata.gtfsvalidator.table.GtfsStopTableContainer;
+import org.mobilitydata.gtfsvalidator.table.GtfsStopTime;
+import org.mobilitydata.gtfsvalidator.table.GtfsTrip;
+
+public class GeospatialUtil {
+  static final int KILOMETER_TO_METER_CONVERSION_FACTOR =
+      1000; // conversion factor from kilometers to meters
+  static final float METER_TO_KILOMETER_CONVERSION_FACTOR =
+      1 / 1000; // conversion factor from kilometers to meters
```

Comment:
You are dividing an integer by integer, so the result is [rounded](https://stackoverflow.com/questions/37795248/integer-division-in-java) and it is 0 (zero). You probably wanted to write `1.0 / 1000` instead.

When I executed your code on a real feed, it produced 8521 errors for 68157 trips, so that looked suspicious.

What was the size and diversity of the feeds that you use? Do you mind to cover that bug with unit tests?

Also, why are you using `float` instead of `double` here?

Reply:
@aababilov we originally tested the validator against 7 datasets. The biggest dataset in terms of `stop_times.txt` is from STM (https://openmobilitydata.org/p/societe-de-transport-de-montreal/39/latest/download) and contains +/- 5.6 millions rows in `stop_times.txt`.

Reply:
The datasets the validator was tested against are listed in the `end_to_end` workflow : https://github.com/MobilityData/gtfs-validator/blob/v1.4.0/.github/workflows/end_to_end.yml

Reply:
Unfortunately, this feed for Montreal was not testing StopTooFarFromTripShapeValidator - its code was never called. The thing is that the Montreal feed has `field_parsing_error` for stop_url field (it uses non-ASCII characters), so parsing failed, the validation process was stopped and StopTooFarFromTripShapeValidator was never invoked.

You can check the generated report.json to find the parsing errors.

```JSON
        {
            "code": "field_parsing_error",
            "notices": [
                {
                    "csvRowNumber": 163,
                    "fieldName": "stop_url",
                    "fieldType": "URL",
                    "fieldValue": "http://www.stm.info/fr/infos/reseaux/metro/square-victoria\u2013oaci",
                    "filename": "stops.txt"
                },
                {
                    "csvRowNumber": 232,
                    "fieldName": "stop_url",
                    "fieldType": "URL",
                    "fieldValue": "http://www.stm.info/fr/infos/reseaux/metro/longueuil\u2013universite-de-sherbrooke",
                    "filename": "stops.txt"
                }
            ],
            "severity": "ERROR",
            "totalNotices": 2
        },
```

Reply:
I wrote https://github.com/MobilityData/gtfs-validator/pull/742 to make clear that parsing failed.

Reply:
In v1.4, the workflows would surface those kind of situation in the GitHub interface by outputting an error, which would in turn lead to all subsequent feeds to be skipped. I wonder if such a process could be replicated.

Reply:
>You are dividing an integer by integer, so the result is rounded and it is 0 (zero). You probably wanted to write 1.0 / 1000 instead.
>Also, why are you using float instead of double here?

Good catch - originally this used a `double` and divided by `1000.0d` as implemented in https://github.com/MobilityData/gtfs-validator/pull/340/files#diff-fd338ec09e9005136186658756b593cc6e7e9917476a0b92a59f27071a1c3314R45. So it should be changed back.